src/JB721StakingDelegate.sol

```diff
@@ -444,6 +467,11 @@ contract JB721StakingDelegate is
                 _leftoverAmount -= _tiers[_i].amount;
             }
 
+            // If there's either a new delegate or old delegate, increase the delegate weight.
+            if (_votingDelegate != address(0)) {
```

Comment:
yeah but if we have a diff address passed in pay to change delegate then we need to consider that right ?
we can just send 0 address once we do it once

Reply:
Ofc, but now if I pay and want to mint 5 nfts and set my delegate to be `address(1)` it will update my delegate to `address(1)` for every nft im minting. When really it just needs to update it once. We can just call `_delegate(..)` in `_processPayment(..)` or in this method outside of the for loop.